### PR TITLE
Update ESLint related packages, support ESLint 6 and 7

### PR DESCRIPTION
**Pull request by paulius-valiunas**
Opening a new PR to replace #320

Most notable differences:
- `@typescript-eslint/explicit-module-boundary-types` has been disabled
- `@bentley/eslint-plugin` now supports both ESLint 6 and 7, so we can continue using it until the CRA fork is updated.

### CODE REVIEW

**Review by grigasp — APPROVED** [review]
approving presentation changes

**Review by calebmshafer — APPROVED** [review]

File: core/common/src/rpc/ModelGeometryChanges.ts [hunk 1]
```diff
@@ -6,12 +6,8 @@
  * @module RpcInterface
  */
 
-import {
-  assert, CompressedId64Set, DbOpcode, GuidString, Id64String,
-} from "@bentley/bentleyjs-core";
-import {
-  Range3d, Range3dProps,
-} from "@bentley/geometry-core";
+import { assert, CompressedId64Set, DbOpcode, GuidString, Id64String } from "@bentley/bentleyjs-core";
```
Comment (pmconne) [inline core/common/src/rpc/ModelGeometryChanges.ts:9]: Did some lint rule require these changes or are they incidental?

Comment (paulius-valiunas) [reply core/common/src/rpc/ModelGeometryChanges.ts:9]: I think the import sorter vscode plugin did this when I saved the file. Do you think this is too much? I wasn't sure if I should reconfigure it to be more forgiving regarding whitespace.

Comment (pmconne) [reply core/common/src/rpc/ModelGeometryChanges.ts:9]: As long as the linter's not going to start hollering about import formatting it's fine with me.

**Review by pmconne — APPROVED** [review]

**Review by bsteinbk — APPROVED** [review]

File: example-code/app/src/backend/RobotWorldRpcImpl.ts [hunk 2]
```diff
@@ -2,12 +2,19 @@
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
-// __PUBLISH_EXTRACT_START__ RpcInterface.implementation
-import { IModelRpcProps, RpcInterface, RpcInterfaceDefinition } from "@bentley/imodeljs-common";
 import { Id64String } from "@bentley/bentleyjs-core";
+// __PUBLISH_EXTRACT_END__
```
Comment (calebmshafer) [inline example-code/app/src/backend/RobotWorldRpcImpl.ts:6]: @paulius-valiunas any idea why this auto-formats to this have the _END_ here?  I tried to switch it locally and every time I saved the file it switched back this state.

**Review by calebmshafer — CHANGES_REQUESTED** [review]

File: example-code/app/src/backend/RobotWorldRpcImpl.ts (hunk 2 above)
Comment (paulius-valiunas) [reply example-code/app/src/backend/RobotWorldRpcImpl.ts]: ah, I didn't notice that. It's the import sorter extension, it wants to have all imports in one place and treats those comments as attached to the imports.

Comment (paulius-valiunas) [reply example-code/app/src/backend/RobotWorldRpcImpl.ts]: I changed it back to the way it was. We probably need to get rid of the sorter extension after all, I don't see a way to keep it from messing up files like this without completely disabling it.

**Comment by paulius-valiunas** [comment]
@calebmshafer could you please look into the extract-api failure?

**Comment by calebmshafer** [comment]
> @calebmshafer could you please look into the extract-api failure?

@paulius-valiunas we are having all sorts of api extractor issues when that file is edited.  I'm trying to address it for @christophermlawson 's changes over in #439.

For this PR I'm going to revert the change to that one file and turn it into a warning at the package level instead of an error.  I'll switch it back once we resolve the issue in the other PR.

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**